system/Database/SQLite3/Table.php

```diff
@@ -240,6 +240,14 @@ protected function createTable()
 
         $this->forge->addField($fields);
 
+        $fieldNames = array_keys($fields);
+
+        $this->keys = array_filter(
+            $this->keys,
+            static fn ($index) => count(array_intersect($index['fields'], $fieldNames))
+            === count($index['fields'])
```

Comment:
This is too long, but it seems better. 

```suggestion
            static fn ($index) => count(array_intersect($index['fields'], $fieldNames)) === count($index['fields'])
```